general/src/types/meta.rs

```diff
@@ -432,6 +432,7 @@ pub struct MetaInner {
 
 impl MetaInner {
     pub fn is_empty(&self) -> bool {
+        // TODO(jan): Evaluate which meta info is relevant enough to keep if the value is gone.
```

Comment:
@mitsuhiko @untitaker This is for a follow-up PR. I would argue that we should always be able to add `original_length` without worrying about serializing a Meta node. If only `original_length` is set but no errors or remarks, then no meta tree should be serialized for this path. This would slim down meta generated for trimmed structures.